lib/ui/screens/album_screen.dart

```diff
@@ -25,14 +28,50 @@ class AlbumScreen extends StatefulWidget {
   State<AlbumScreen> createState() => _AlbumScreenState();
 }
 
-class _AlbumScreenState extends State<AlbumScreen> {
+class _AlbumScreenState extends State<AlbumScreen>
+    with TickerProviderStateMixin {
   late final AlbumCubit _albumCubit;
+  bool galleryShown = false;
+  int initialGalleryIndex = 0;
+
+  late AnimationController filledController;
+  late Animation<double> filledAnimation;
+
+  late AnimationController controller;
+  late Animation<double> animation;
+
+  PageController pageController = PageController(initialPage: 0);
+  PageController pageController2 = PageController(initialPage: 0);
+
+  void _onMainScroll() {
```

Comment:
`_onMainScroll` sounds like the gridviews vertical scroll. Maybe `_onGalleryScroll` is more clear, and a docstring/comment would help of course